packages/@aws-cdk/aws-redshift/lib/cluster.ts

```diff
@@ -592,4 +628,24 @@ export class Cluster extends ClusterBase {
       return nodeCount;
     }
   }
+
+  public addToParameterGroup(name: string, value: string): AddParameterResult {
+    if (!this.parameterGroup && this.autoCreateParameterGroup) {
+      const param: { [name: string]: string } = {};
+      param[name] = value;
+      this.parameterGroup = new ClusterParameterGroup(this, 'ParameterGroup', {
+        description: `Parameter Group for the ${this.clusterIdentifier} Redshift cluster`,
+        parameters: param,
+      });
+      return { parameterAddedResult: AddParameterResultStatus.SUCCESS };
+    } else if (this.parameterGroup) {
+      return this.parameterGroup.addParameter(name, value);
+    } else {
+      return { parameterAddedResult: AddParameterResultStatus.IMPORTED_RESOURCE_FAILURE };
```

Comment:
Why not just throw an error instead of returning a failure?

Reply:
Please throw errors instead of returning failures. However, I still don't understand why these operations default to failures. What about the resource being imported prevents it from succeeding?

Regardless of why it fails for imported resources, the import should not be considered the default behavior anyway. Imports are the 'special-ish' case.